packages/engine/lib/orchestrator/src/experiment.rs

```diff
@@ -95,6 +95,25 @@ pub struct ExperimentConfig {
     clap(global = true, short = 'w', long, default_value_t = num_cpus::get(), validator = at_least_one, env = "HASH_WORKERS")
     )]
     pub num_workers: usize,
+
+    #[cfg_attr(feature = "clap", clap(global = true, long, default_value_t = 0))]
+    /// Initial size of the V8 heap.
+    ///
+    /// Setting this value allows to avoid garbage collection while the heap is small enough.
+    pub v8_initial_heap_constraint: usize,
+
+    #[cfg_attr(
+        feature = "clap",
+        clap(global = true, long, default_value_t = 4_000_000_000)
```

Comment:
i chose 4GB since it's about what is available with pointer compression.
I also don't think a lot of people will try to run the engine with too little memory and in that case they are probably aware of their small memory space.